src/main/scala/workflow/Optimizer.scala

```diff
@@ -0,0 +1,37 @@
+package workflow
+
+/**
+ * Optimizes a Pipeline DAG
+ */
+object Optimizer extends RuleExecutor {
+  protected val batches: Seq[Batch] = Batch("DAG Optimization", FixedPoint(100), EquivalentNodeMerger) :: Nil
```

Comment:
Just curious -- for the pipelines we have today, how useful is merging equivalent nodes ?